src/test/java/springboot/webapp/usersmanager/services_tests/UserServiceTests.java

```diff
@@ -0,0 +1,144 @@
+package springboot.webapp.usersmanager.services_tests;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.web.servlet.*;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import springboot.webapp.usersmanager.controllers.UserController;
+import springboot.webapp.usersmanager.entities.Role;
+import springboot.webapp.usersmanager.entities.User;
+import springboot.webapp.usersmanager.repositories.UserRepository;
+import springboot.webapp.usersmanager.services.UserService;
+
+import java.util.*;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(UserController.class)
+public class UserServiceTests {
+    @Autowired
+    MockMvc mockMvc;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @MockBean
+    UserRepository userRepository;
+
+    @MockBean
+    UserService userService;
+
+    User user1 = new User(1, "Sergey", "Gonchar", "sergey@gmail.com", Role.ROLE_ADMIN);
+
+    User user2 = new User(2, "Sergey", "Danilov", "danilov@gmail.com", Role.ROLE_USER);
+
+    User user3 = new User(3, "Ivan", "Petrov", "petrov@gmail.com", Role.ROLE_USER);
+
+    @Test
+    public void getAll_usersList_ok() throws Exception {
+        List<User> users = new ArrayList<>(Arrays.asList(user1, user2, user3));
+
+        when(userService.getAll()).thenReturn(users);
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/users")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(3)));
```

Comment:
use AssertThat

Reply:
Assert.assertThat is deprecated
![image](https://user-images.githubusercontent.com/83033399/144210644-da8a0d38-c736-40b6-b79a-a70d3e6557c9.png)